compute/resource_live_table.go

```diff
@@ -0,0 +1,221 @@
+}
+
+type Filters struct {
+	Include []string `json:"include,omitempty"`
+	Exclude []string `json:"exclude,omitempty"`
+}
+
+type PipelineSpec struct {
+	Id            string            `json:"id,omitempty"`
+	Name          string            `json:"name,omitempty"`
+	Storage       string            `json:"storage,omitempty"`
+	Configuration map[string]string `json:"configuration,omitempty"`
+	Clusters      []PipelineCluster `json:"clusters,omitempty"`
+	Libraries     []PipelineLibrary `json:"libraries,omitempty"`
+	Filters       []Filters         `json:"filters"`
+	Continuous    bool              `json:"continuous,omitempty"`
+}
+
+type PipelineState string
+
+const (
+	StatusDeploying  PipelineState = "DEPLOYING"
+	StatusStarting   PipelineState = "STARTING"
+	StatusRunning    PipelineState = "RUNNING"
+	StatusStopping   PipelineState = "STOPPPING"
+	StatusDeleted    PipelineState = "DELETED"
+	StatusRecovering PipelineState = "RECOVERING"
+	StatusFailed     PipelineState = "FAILED"
+	StatusResetting  PipelineState = "RESETTING"
+	StatusIdle       PipelineState = "IDLE"
+)
+
+type PipelineHealthStatus string
+
+const (
+	HealthStatusHealthy   PipelineHealthStatus = "HEALTHY"
+	HealthStatusUnhealthy PipelineHealthStatus = "UNHEALTHY"
+)
+
+type PipelineInfo struct {
+	PipelineId string                `json:"pipeline_id"`
+	Spec       *PipelineSpec         `json:"spec"`
+	State      *PipelineState        `json:"state"`
+	Cause      string                `json:"cause"`
+	ClusterId  string                `json:"cluster_id"`
+	Name       string                `json:"name"`
+	Health     *PipelineHealthStatus `json:"health"`
+}
+
+// Live Tables API Client
+
+type LiveTablesAPI struct {
+	client *common.DatabricksClient
+	ctx    context.Context
+}
+
+type LiveTableId string
+
+func NewLiveTablesAPI(ctx context.Context, m interface{}) LiveTablesAPI {
+	return LiveTablesAPI{m.(*common.DatabricksClient), ctx}
+}
+
+func (a LiveTablesAPI) Create(s PipelineSpec, allowDuplicateNames bool) (LiveTableId, error) {
+	var id LiveTableId
+	err := a.client.Post(a.ctx, "/pipelines", map[string]interface{}{
+		"spec":                  s,
+		"allow_duplicate_names": allowDuplicateNames,
+	}, &id)
+	return id, err
+}
+
+func (a LiveTablesAPI) Read(id LiveTableId) (p PipelineInfo, err error) {
+	err = a.client.Get(a.ctx, "/pipelines/"+string(id), nil, &p)
+	return
+}
+
+func (a LiveTablesAPI) Update(id LiveTableId, s PipelineSpec, allowDuplicateNames bool) error {
+	return a.client.Put(a.ctx, "/pipelines/"+string(id), map[string]interface{}{
+		"spec":                  s,
+		"allow_duplicate_names": allowDuplicateNames,
+	})
+}
+
+func (a LiveTablesAPI) Delete(id LiveTableId) error {
+	return a.client.Delete(a.ctx, "/pipelines/"+string(id), nil)
+}
+
+// Live Tables Resource Definition
+
+var pipelineSchema = common.StructToSchema(PipelineSpec{}, func(m map[string]*schema.Schema) map[string]*schema.Schema { return m })
```

Comment:
we probably should define it inside of `ResourceLiveTable` function scope, as schema is not used elsewhere in the package.

Reply:
Good suggestion :)